public/app/features/dashboard/containers/PublicDashboardPage.tsx

```diff
@@ -1,13 +1,93 @@
+import { DashNavTimeControls } from '../components/DashNav/DashNavTimeControls';
+import { DashboardFailed } from '../components/DashboardLoading/DashboardFailed';
+import { DashboardLoading } from '../components/DashboardLoading/DashboardLoading';
+import { PublicDashboardFooter } from '../components/PublicDashboardFooter/PublicDashboardsFooter';
+import { DashboardGrid } from '../dashgrid/DashboardGrid';
+import { getTimeSrv } from '../services/TimeSrv';
+import { DashboardModel } from '../state';
+import { initDashboard } from '../state/initDashboard';
+
+import { DashboardPageRouteParams, DashboardPageRouteSearchParams } from './DashboardPage';
 
 export type Props = GrafanaRouteComponentProps<DashboardPageRouteParams, DashboardPageRouteSearchParams>;
 
+const Toolbar = ({ dashboard }: { dashboard: DashboardModel }) => {
+  const dispatch = useDispatch();
+
+  const onChangeTimeZone = (timeZone: TimeZone) => {
+    dispatch(updateTimeZoneForSession(timeZone));
+  };
+
+  return (
+    <PageToolbar title={dashboard.title}>
+      <ToolbarButtonRow alignment="right">
+        {!dashboard.timepicker.hidden && (
+          <DashNavTimeControls dashboard={dashboard} onChangeTimeZone={onChangeTimeZone} />
+        )}
+      </ToolbarButtonRow>
+    </PageToolbar>
+  );
+};
+
 const PublicDashboardPage = (props: Props) => {
-  return <DashboardPage isPublic {...props} />;
+  const { match, route, location } = props;
+  const dispatch = useDispatch();
+  const context = useGrafana();
+  const prevProps = usePrevious(props);
+  const dashboardState = useSelector((store) => store.dashboard);
+  const dashboard = dashboardState.getModel();
+
+  useEffect(() => {
+    dispatch(
+      initDashboard({
+        routeName: route.routeName,
+        fixUrl: false,
+        accessToken: match.params.accessToken,
+        keybindingSrv: context.keybindings,
+      })
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    if (prevProps?.location.search !== location.search) {
+      const prevUrlParams = prevProps?.queryParams;
+      const urlParams = props.queryParams;
+
+      if (urlParams?.from !== prevUrlParams?.from || urlParams?.to !== prevUrlParams?.to) {
+        getTimeSrv().updateTimeRangeFromUrl();
+      }
+
+      if (!prevUrlParams?.refresh && urlParams?.refresh) {
+        getTimeSrv().setAutoRefresh(urlParams.refresh);
+      }
+    }
+  }, [prevProps, location.search, props.queryParams]);
+
+  if (!dashboard) {
+    return <DashboardLoading initPhase={dashboardState.initPhase} />;
+  }
+
+  return (
+    <>
+      <Page layout={PageLayoutType.Canvas} toolbar={<Toolbar dashboard={dashboard} />}>
+        {dashboardState.initError && <DashboardFailed />}
+        <DashboardGrid dashboard={dashboard} isEditable={false} viewPanel={null} editPanel={null} />
+      </Page>
+      <PublicDashboardFooter />
```

Comment:
better to put this in header I think next to dashboard title

Reply:
@torkelo this feature was discussed with the UX team and we ended up putting it as a floating footer. Should we discuss it again?

Reply:
yea, but then you did not have a header, right? if we have a header (with title) seems make more sense to put it there.

Reply:
but just a suggestion